srml/support/src/dispatch.rs

```diff
@@ -615,6 +615,132 @@ macro_rules! decl_module {
 		}
 	};
 
+	// Declare a `Call` variant parameter that should be encoded `compact`.
+	(@decl_call_variant_params
+		$( #[$attr:meta] )*
+		$call_type:ident;
+		<$trait_instance:ident: $trait_name:ident$(<I>, $instance:ident: $instantiable:path $(= $module_default_instance:path)?)?>
+		{ $( $generated_variants:tt )* }
+		{ $( $current_params:tt )* }
+		variant $fn_name:ident;
+		$( #[doc = $doc_attr:tt] )*
+		#[compact]
+		$type:ty;
+		$( $rest:tt )*
+	) => {
+		$crate::decl_module! {
+			@decl_call_variant_params
+			$( #[$attr] )*
+			$call_type;
+			<$trait_instance: $trait_name $(<I>, $instance: $instantiable $(= $module_default_instance)? )?>
+			{ $( $generated_variants )* }
+			{
+				$( $current_params )*
+				#[codec(compact)]
+				$type,
+			}
+			variant $fn_name;
+			$( $rest )*
+		}
+	};
+
+	// Declare a `Call` variant parameter.
+	(@decl_call_variant_params
+		$( #[$attr:meta] )*
+		$call_type:ident;
+		<$trait_instance:ident: $trait_name:ident$(<I>, $instance:ident: $instantiable:path $(= $module_default_instance:path)?)?>
+		{ $( $generated_variants:tt )* }
+		{ $( $current_params:tt )* }
+		variant $fn_name:ident;
+		$(#[doc = $doc_attr:tt])*
+		$type:ty;
+		$( $rest:tt )*
+	) => {
+		$crate::decl_module! {
+			@decl_call_variant_params
+			$( #[$attr] )*
+			$call_type;
+			<$trait_instance: $trait_name $(<I>, $instance: $instantiable $(= $module_default_instance)? )?>
+			{ $( $generated_variants )* }
+			{
+				$( $current_params )*
+				$type,
+			}
+			variant $fn_name;
+			$( $rest )*
+		}
+	};
+
+	(@decl_call_variant_params
+		$( #[$attr:meta] )*
+		$call_type:ident;
+		<$trait_instance:ident: $trait_name:ident$(<I>, $instance:ident: $instantiable:path $(= $module_default_instance:path)?)?>
+		{ $( $generated_variants:tt )* }
+		{ $( $current_params:tt )* }
+		variant $fn_name:ident;
+		$(#[doc = $doc_attr:tt])*
+		variant $next_fn_name:ident;
+		$( $rest:tt )*
+	) => {
+		$crate::decl_module! {
+			@decl_call_variant_params
+			$( #[$attr] )*
+			$call_type;
+			<$trait_instance: $trait_name $(<I>, $instance: $instantiable $(= $module_default_instance)? )?>
+			{
+				$( $generated_variants )*
+				#[allow(non_camel_case_types)]
+				$(#[doc = $doc_attr])*
+				$fn_name (
+					$( $current_params )*
+				),
+			}
+			{}
+			variant $next_fn_name;
+			$( $rest )*
+		}
+	};
+
+	(@decl_call_variant_params
+		$( #[$attr:meta] )*
+		$call_type:ident;
+		<$trait_instance:ident: $trait_name:ident$(<I>, $instance:ident: $instantiable:path $(= $module_default_instance:path)?)?>
+		{ $( $generated_variants:tt )* }
+		{ $( $current_params:tt )* }
+		variant $fn_name:ident;
+		$( #[doc = $doc_attr:tt] )*
```

Comment:
I understand you want to use the parsing ahead of time because of the previous arm. But I think that could be avoid either by encapsulating variant params in {} on the first call of decl_call_variant_params, or by just matching everything after the current variant line 681.
It seems possible to me no ?
then both this arm and the arm below can be merged

Reply:
not sure that make sense but I think there is a way to just impl on this arm:
```rust
(@decl_call_variant_params
		$( #[$attr:meta] )*
		$call_type:ident;
		<$trait_instance:ident: $trait_name:ident$(<I>, $instance:ident: $instantiable:path $(= $module_default_instance:path)?)?>
		{ $(generated)* }
		{}
```